lib/drm/drm_engine.js

```diff
@@ -2668,6 +2637,47 @@ shaka.drm.DrmEngine = class {
     this.newInitData('cenc', combinedData);
     return this.allSessionsLoaded_;
   }
+
+  /**
+   * Create a DrmInfo using configured clear keys and assign it to each variant.
+   * Only modify variants if clear keys have been set.
+   *
+   * @param {!Object.<string, string>} configClearKeys
+   * @param {!Array.<shaka.extern.Variant>} variants
+   */
+  static configureClearKey(configClearKeys, variants) {
+    /**
+     * Create a DrmInfo using configured clear keys.
+     * The server URI will be a data URI which decodes to a clearkey license.
+     * @return {?shaka.extern.DrmInfo} or null if clear keys are not configured.
+     * @see https://bit.ly/2K8gOnv for the spec on the clearkey license format.
+     */
+    const getClearKeyDrmInfo = () => {
```

Comment:
I don't really see much of a reason to define a helper method only to immediately call it exactly once. Couldn't you just fold this into the main method, and return early if `clearKeys.size == 0`?

Reply:
Removed